lib/msal-common/src/logger/Logger.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { StringUtils } from "../utils/StringUtils";
+import pkg from "../../package.json";
+
+export type LoggerMessageOptions = {
+    logLevel: LogLevel,
+    correlationId?: string,
+    containsPii?: boolean
+};
+
+export enum LogLevel {
+    Error,
+    Warning,
+    Info,
+    Verbose
+};
+
+export interface ILoggerCallback {
+    (level: LogLevel, message: string, containsPii: boolean): void;
+}
+
+export class Logger {
+
+    /**
+     * @hidden
+     */
+    private correlationId: string;
+
+    /**
+     * @hidden
+     */
+    private level: LogLevel = LogLevel.Info;
+
+    /**
+     * @hidden
+     */
+    private piiLoggingEnabled: boolean;
+
+    /**
+     * @hidden
+     */
+    private localCallback: ILoggerCallback;
+
+    constructor(localCallback: ILoggerCallback, piiLoggingEnabled: boolean) {
+        this.localCallback = localCallback;
+        this.piiLoggingEnabled = piiLoggingEnabled;
+    }
+
+    /**
+     * @hidden
+     */
+    private logMessage(logMessage: string, options: LoggerMessageOptions): void {
+        if ((options.logLevel > this.level) || (!this.piiLoggingEnabled && options.containsPii)) {
+            return;
+        }
+        const timestamp = new Date().toUTCString();
+        let logHeader: string = StringUtils.isEmpty(this.correlationId) ? `[${timestamp}] : ` : `[${timestamp}] : [${this.correlationId}]`;
+        const log = `${logHeader} : ${pkg.version} : ${LogLevel[options.logLevel]} - ${logMessage}`;
+        this.executeCallback(options.logLevel, log, options.containsPii);
+    }
+
+    /**
+     * @hidden
+     */
+    executeCallback(level: LogLevel, message: string, containsPii: boolean) {
+        if (this.localCallback) {
+            this.localCallback(level, message, containsPii);
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    error(message: string, correlationId?: string): void {
+        this.logMessage(message, {
+            logLevel: LogLevel.Error,
+            containsPii: false,
+            correlationId: correlationId || ""
```

Comment:
Is `correlationId` used? And I would argue to have this fallback logic to empty string inside of `logMessage`, so that you don't have to do it in each log level function.